sync.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+cwd=$(pwd)
+if [ ! -f sync ]; then
+    go get -u -f github.com/dkonovenschi/aggretastic-sync
+    cd $GOPATH/src/github.com/dkonovenschi/aggretastic-sync
+    go build -o sync
+    cp sync "$cwd"
+    cd $cwd
+fi
+
+go get github.com/olivere/elastic
+./sync
```

Comment:
A general comment on `aggretastic-sync`:

1. the panic-on-error approach is used, that's is not so good for user. It's better to use errors and pop them up.
Some errors are ok to be just logged, skip an item and continue. Some errors will stop the whole things.
The sync command will be called by final user of library, and if user gets a panic, usually it means `it's not user's fault, it's some code problem`

2. The actual filesystem is used for temporary files, and by default the global `/tmp` folder is used. It has drawbacks.
Few times I had `tmp` was not cleaned after executing (although I saw there is a defer with cleaning), and then, when you retry `sync` (and the tmp folder already exists) - you can get a panic on lots of things

Real filesystem cause you to use lots of `os.` things like reading/writing/ls/mkdir etc. It may cause bugs depending on what OS user uses, what rights he has on specific folders, etc, etc.

Maybe consider using in-memory-storage? the `git` library has option to git clone in a memory-storage. It shares the basic `Reader` interface. AST supports the same functions from `[]byte` source. instead of extra temporary files you can use buffers that supports the same Read/Write files and stores them in a slice instead of real directory.